docs/contributing/coding-style-guide.md

```diff
@@ -288,6 +288,65 @@ in future. There was even Go 2 proposal for [disabling this in the language itse
 
 Similar to this problem is the package name shadowing. While it is less dangerous, it can cause similar issues, so avoid package shadowing if you can.
 
+#### Avoid Wrapping nil With Error Message; Avoid Passing nil to MultiError
+
+It's very tempting to use "magic" functionality of e.g errors.Wrap that returns nil if wrapped error equals nil as it created more compacted code.
+Avoid that. Consider following case:
+
+<table>
+<tbody>
+<tr><th>Avoid 🔥</th></tr>
+<tr><td>
+
+```go
+  	if !replicas[endpoint].replicated && h.options.ReplicationFactor > 1 {
```

Comment:
I think this example is way too complicated to illustrate the point of returning a wrapped error. it distracts from the point

Reply:
agree

Reply:
Yes. I have a hard time figuring out what's happening too. Still haven't found it.

A nice idea for such examples would also be to use diffs.

Essentially like this, to give an example. Obviously we lose Go syntax highlighting. Not sure what's better.
```diff
func example() error {
-	return foo()
+	if err := foo(); err != nil {
+		return err
+	}
+	return nil
```

Reply:
Simplified

Reply:
diff would be nice, but for consistency let's keep it simple for now